dev/functional/static_magic.h

```diff
@@ -35,6 +61,12 @@ namespace sqlite_orm {
         decltype(auto) static_if(T&& t) {
             return static_if(std::integral_constant<bool, B>{}, std::forward<T>(t), empty_callable());
         }
+
+        template<bool B, typename L, typename... Args>
+        void call_if_constexpr(L&& lambda, Args&&... args) {
+            static_if<B>(static_cast<L&&>(lambda))(static_cast<Args&&>(args)...);
```

Comment:
why don't you use `move` instead of `static_cast<T&&>`?

Reply:
The arguments are just forwarded and comes w/o the cost of including <utility> for `std::forward`.
But it seems to cause confusion?

Reply:
up to you

Reply:
Well, it's causing confusion, otherwise you wouldn't ask :) I just use the static cast because it's an internal implementation detail that even the STL guys at Microsoft use.
But you do have to know what the argument is, and `forward` expresses intent.

Reply:
I changed my mind. Let's place an intent here also. Many developers know about forward and move but not many developers know that these functions contain static cast inside

Reply:
Agreed. I doubt the "many developers" part, because to know about forwarding you need to know about "universal references" and about how forwarding works. But there's no question that programming is about expressing intent, which makes things easier to read in an already complex piece of work!